core/cold-wallet/wallet/cold-wallet.ts

```diff
@@ -62,6 +63,30 @@ export class ColdWallet {
     return new CTransactionSegWit(tx).toHex();
   }
 
+  public async signTx(data: SignTxPayload): Promise<string> {
+    if (!this.wallet) throw new Error('Wallet is not initialized');
```

Comment:
Do we need some TX validation here? Not all TXs are allowed 🤔

Reply:
Yes, currently implementing it within its own ticket